File: GestionStagiaires/src/main/java/com/GestionStagiaires/GestionStagiaires/Model/Encadrant.java
package com.GestionStagiaires.GestionStagiaires.Model;

import com.GestionStagiaires.GestionStagiaires.Enum.EncadrantType;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Entity
@AllArgsConstructor
@NoArgsConstructor
@Setter
@Getter
public class Encadrant {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long encadrantId;
    private String nom;
    private String prenom;
    private String email;
    private String telephone;
    @Enumerated(EnumType.STRING)
    private EncadrantType encadrantType;

    @OneToMany(mappedBy = "encadrant")
    private List<Stage> stagesEncadres;
}

File: gestionStagiares_Front/src/app/Model/encadrant.ts
// encadrant.model.ts

import { Stage } from "./stage";


export class Encadrant {
  encadrantId!: number;
  nom!: string;
  prenom!: string;
  email!: string;
  telephone!: string;
  encadrantType!: string;

  stagesEncadres?: Stage[];
}

